compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h

```diff
@@ -117,12 +117,19 @@ SmallVector<Value> getTileSizes(OpBuilder &b, Operation *op, unsigned level);
 /// Sets the lowering configuration, overwriting existing attribute values.
 void setLoweringConfig(Operation *op, Attribute config);
 
+/// Sets an attribute to identify the rootOp and adds any information needed for
+/// the tuner from compiler. Currently, only sets a `UnitAttr`. Note that this
+/// attribute is not used by the compiler at any level and is only intended for
+/// tuner use.
+void setRootOpInfo(Operation *op);
+
 /// Convenience function that sets the lowering configuration on the operation
 /// and translation info.
 inline LogicalResult setOpConfigAndEntryPointFnTranslation(
     mlir::FunctionOpInterface entryPointFn, Operation *op,
     IREE::Codegen::LoweringConfigAttrInterface config,
     IREE::Codegen::TranslationInfoAttr translationInfo) {
+  setRootOpInfo(op);
```

Comment:
To be more specific -- If it were me, I'd add a flag in this file and wrap this statement with the flag.